### DISCUSSION TITLE
imcoding 代码泛型求助

### DISCUSSION THREAD

**Discussion by Demian101** [NonceGeek/Move-Camp#237] [Q&A] [answered]
高级 - 实现泛型：
<img width="1202" alt="image" src="https://user-images.githubusercontent.com/33189338/212018075-bc2103c2-7cbb-462b-a10b-668487fa1ed6.png">

不太知道怎么改了

**Comment by xiegudong45** [comment, marked as answer]
能力补充应该卸载函数名后面的<T>中，我的理解是get_with_default<T>的<T>是一个“入参”，只有这里定义了范型后面入参 & 函数体中才能使用范型～
![image](https://user-images.githubusercontent.com/30063330/212022829-a4ea2756-3dc7-4a5c-a07c-940edda44347.png)

**Comment by Demian101** [comment]
use Std::Vector;

struct Option<T> has drop {
  vec: vector<T>
}

public fun get_with_default<T: drop+copy>(t: &Option<T>, default: T): T {
  if (Vector::is_empty(&t.vec)) {
    default
  } else {
    *Vector::borrow(&t.vec, 0)
  }
}

过了,感谢大佬
